test/lib/jobmodel.test.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+const assert = require('chai').assert;
+const mockery = require('mockery');
+const sinon = require('sinon');
+
+sinon.assert.expose(assert, { prefix: '' });
+
+describe('Job Model', () => {
+    let JobModel;
+    let datastore;
+    let hashaMock;
+    let job;
+    const jobData = {
+        id: 'd398fb192747c9a0124e9e5b4e6e8e841cf8c71c',
+        pipelineId: '151c9b11e4a9a27e9e374daca6e59df37d8cf00f',
+        name: 'deploy',
+        state: 'ENABLED',
+        triggers: [],
+        triggeredBy: ['151c9b11e4a9a27e9e374daca6e59df37d8cf00f']
+    };
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    beforeEach(() => {
+        datastore = {
+            get: sinon.stub(),
+            scan: sinon.stub(),
+            update: sinon.stub(),
+            save: sinon.stub()
+        };
+        hashaMock = {
+            sha1: sinon.stub()
+        };
+        mockery.registerMock('screwdriver-hashr', hashaMock);
+
+        // eslint-disable-next-line global-require
+        JobModel = require('../../lib/jobmodel');
+
+        job = new JobModel(datastore);
+    });
+
+    afterEach(() => {
+        datastore = null;
+        mockery.deregisterAll();
+        mockery.resetCache();
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    it('constructs', () => {
+        const result = new JobModel(datastore);
+
+        assert.isOk(result);
+    });
+
+    it('has the correct API', () => {
```

Comment:
The other tests infer that the public methods exist in order to assert their functionality. Do we need this specific test?